closed/adds/jdk/src/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -0,0 +1,1073 @@
+                                if (in.length != 2) {
+                                    printStackTraceAndExit(
+                                            "Constraint attributes format is incorrect: " + providerInfo);
+                                }
+                            }
+                        }
+
+                        Constraint constraint = new Constraint(inType, inAlgorithm, inAttributes);
+
+                        if (debug != null) {
+                            debug.println("Loading constraints for security provider: " + providerName
+                                    + " with constraints type: " + inType
+                                    + " algorithm: " + inAlgorithm
+                                    + " attributes: " + inAttributes);
+                        }
+                        constraints[cNum] = constraint;
+                        cNum++;
+                    }
+                    providerConstraints.put(providerName, constraints);
+                    if (debug != null) {
+                        debug.println("Loaded constraints for security provider: " + providerName);
+                    }
+                } else {
+                    printStackTraceAndExit("Constraint format is incorrect: " + providerInfo);
+                }
+            }
+        }
+
+        /**
+         * Check if the Service is allowed in restricted security mode.
+         *
+         * @param service the Service to check
+         * @return true if the Service is allowed
+         */
+        boolean isRestrictedServiceAllowed(Service service) {
+            String providerName = service.getProvider().getName();
+
+            // Provider with argument, remove argument.
+            // e.g. SunPKCS11-NSS-FIPS, remove argument -NSS-FIPS.
+            int pos = providerName.indexOf('-');
+            providerName = (pos < 0) ? providerName : providerName.substring(0, pos);
+
+            Constraint[] constraints = providerConstraints.get(providerName);
+
+            if (constraints == null) {
+                // Disallow unknown providers.
+                return false;
+            } else if (constraints.length == 0) {
+                // Allow this provider with no constraints.
+                return true;
+            }
+
+            // Check the constraints of this provider.
+            String type = service.getType();
+            String algorithm = service.getAlgorithm();
+
+            for (Constraint constraint : constraints) {
+                String cType = constraint.type;
+                String cAlgorithm = constraint.algorithm;
+                String cAttribute = constraint.attributes;
+
+                if (!isAsterisk(cType) && !type.equalsIgnoreCase(cType)) {
+                    // The constraint doesn't apply to the service type.
+                    continue;
+                }
+                if (!isAsterisk(cAlgorithm) && !algorithm.equals(cAlgorithm)) {
```

Comment:
Right now, it doesn’t affect any functions. Because the algorithms from the java.security restricted security provider constraint list, are exactly same as the ones which registered in the provider java codes. But, since the type is using “equalsIgnoreCase()”, so I would like to use the same “equalsIgnoreCase()” for the algorithms, to make them look the same(by using the same equalsIgnoreCase() method). So, I will create a PR on OpenJDK Next first and then backport this change back.